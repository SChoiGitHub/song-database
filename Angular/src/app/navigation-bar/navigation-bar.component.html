<ul ngbNav #nav="ngbNav" class="nav-pills">
  <li ngbNavItem [disabled]="true" class="">
    <a ngbNavLink>Song Database</a>
  <li>
  <app-navigationitem link="/" label="View Songs"></app-navigationitem>
  <app-navigationitem link="/addSong" label="Add Song"></app-navigationitem>
</ul>
