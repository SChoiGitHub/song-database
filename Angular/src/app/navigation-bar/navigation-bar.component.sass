.container
    display: flex
    padding: 16px

.flexItem
    padding: 0 8px 0 0
    align-self: center

.ignoreFirst
  color: inherit
  background-color: transparent
  text-decoration: none
  box-shadow: none