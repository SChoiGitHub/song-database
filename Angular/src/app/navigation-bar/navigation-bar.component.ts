import { Component, ViewChild } from '@angular/core';
import { NgbNav } from '@ng-bootstrap/ng-bootstrap';

@Component({
  selector: '.navigationBar',
  templateUrl: './navigation-bar.component.html',
  styleUrls: ['./navigation-bar.component.sass']
})
export class NavigationBarComponent {
  @ViewChild('nav', { static: true }) nav!: NgbNav;

  ngOnInit() {
    this.nav.select(-1);
  }
}
