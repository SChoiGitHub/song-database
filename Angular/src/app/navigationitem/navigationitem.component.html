<li ngbNavItem>
  <button [routerLink]="link" ngbNavLink [attr.aria-label]="'Navigate to ' + label" (onClick)="setToActive()"
    [ngClass]="class">
    {{label}}
  </button>
</li>
