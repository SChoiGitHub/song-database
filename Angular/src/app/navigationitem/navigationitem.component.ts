import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-navigationitem',
  templateUrl: './navigationitem.component.html',
  styleUrls: ['./navigationitem.component.sass']
})
export class NavigationitemComponent {
  @Input() link!: string;
  @Input() label!: string;
  class: string = '';

  setToActive() {
    this.class = 'active';
  }
}
