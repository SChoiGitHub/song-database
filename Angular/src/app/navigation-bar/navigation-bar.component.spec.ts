import { expect } from '@jest/globals';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NavigationBarComponent } from './navigation-bar.component';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { NavigationitemComponent } from '../navigationitem/navigationitem.component';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';

describe('NavigationBarComponent', () => {
  let component: NavigationBarComponent;
  let fixture: ComponentFixture<NavigationBarComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [NavigationBarComponent, NavigationitemComponent],
      imports: [RouterTestingModule, NgbModule],
    });
    fixture = TestBed.createComponent(NavigationBarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });


  it('should be active for the active link', () => {
    //Act
    let button = fixture.debugElement.query(By.css('button[aria-label="Navigate to Add Song"]'));
    button.nativeElement.click();
    fixture.detectChanges();

    //Assert
    expect(button.nativeElement).toMatchSnapshot();
    expect(button.nativeElement.class).toEqual('active');
  });
});
