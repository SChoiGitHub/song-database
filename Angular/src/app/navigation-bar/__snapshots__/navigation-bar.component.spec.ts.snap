// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarComponent should be active for the active link 1`] = `
<button
  aria-controls="ngb-nav-5-panel"
  aria-disabled="false"
  aria-label="Navigate to Add Song"
  aria-selected="true"
  class="nav-link active"
  ngbnavlink=""
  role="tab"
  type="button"
>
   Add Song 
</button>
`;

exports[`NavigationBarComponent should match snapshot 1`] = `
<div
  id="root0"
>
  <ul
    class="nav-pills nav"
    ngbnav=""
    role="tablist"
  >
    <li
      class="nav-item"
      ngbnavitem=""
      role="presentation"
    >
      <a
        aria-controls="ngb-nav-0-panel"
        aria-disabled="true"
        aria-selected="true"
        class="nav-link active disabled"
        href=""
        ngbnavlink=""
        role="tab"
        tabindex="-1"
      >
        Song Database
      </a>
    </li>
    <li>
      <app-navigationitem
        label="View Songs"
        link="/"
      >
        <li
          class="nav-item"
          ngbnavitem=""
          role="presentation"
        >
          <button
            aria-disabled="false"
            aria-label="Navigate to View Songs"
            aria-selected="false"
            class="nav-link"
            ngbnavlink=""
            role="tab"
            type="button"
          >
             View Songs 
          </button>
        </li>
      </app-navigationitem>
      <app-navigationitem
        label="Add Song"
        link="/addSong"
      >
        <li
          class="nav-item"
          ngbnavitem=""
          role="presentation"
        >
          <button
            aria-disabled="false"
            aria-label="Navigate to Add Song"
            aria-selected="false"
            class="nav-link"
            ngbnavlink=""
            role="tab"
            type="button"
          >
             Add Song 
          </button>
        </li>
      </app-navigationitem>
    </li>
  </ul>
</div>
`;
